### PR TITLE
Remove duplicate raw to transaction code and replace with helper

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

**Comment by fluidvanadium** [comment]
good and hoping i can merge first

File: zingolib/src/lightclient/test_features.rs [hunk 1]
```diff
@@ -85,4 +82,15 @@ impl LightClient {
...
+
+    /// Creates a [`zcash_client_backend::zip321::TransactionRequest`] from rust primitives for simplified test writing.
+    pub fn transaction_request_from_send_inputs(
+        &self,
+        address_amount_memo_tuples: Vec<(&str, u64, Option<&str>)>,
+    ) -> TransactionRequest {
+        let receivers =
+            receivers_from_send_inputs(address_amount_memo_tuples, &self.config().chain);
+        transaction_request_from_receivers(receivers)
+            .expect("should be able to create a transaction request as receivers are valid.")
```
Comment (fluidvanadium) [inline zingolib/src/lightclient/test_features.rs]: can we return an error value instead of expecting? its possible to get an error here.

**Review by fluidvanadium — DISMISSED** [review]
changing u32 to u64 is correct. i forgot to factor in the satoshis when i did the calculation.

File: zingolib/src/lightclient/test_features.rs (hunk 1 above)
Comment (fluidvanadium) [reply zingolib/src/lightclient/test_features.rs]: valid receivers can still result in invalid request in some cases.

Comment (Oscar-Pepper) [reply zingolib/src/lightclient/test_features.rs]: I dont have an issue with test only functions containing expects. if for example there were too many payments in a test, then a panic here would be enough information to correct the test. anyways I have added the return of the zip321 error for you

**Review by zancas — APPROVED** [review]

**Review by fluidvanadium — APPROVED** [review]
i like having transaction_request_from_send_inputs

**Comment by fluidvanadium** [comment]
re-opened as #1078

**Pull request merged**